### ISSUE TITLE
thread_id vs thid inconsistencies

### ISSUE THREAD

**Issue by brianorwhatever** [decentralized-identity/didcomm-messaging#223]
Both of these names are used throughout the spec I assume we want to standardize on one. Which one?

**Comment by dhh1128** [comment]
I don't care that much. Historically, `thid` comes from DIDComm 1, and people who wrote some early DIDComm v2 materials decided they wanted greater clarity, so they started using `thread_id`. But there was never any vote or consensus on it.

The two things I would note are:

1. We should probably articulate best practices for header names, and make our decision consistent with what we say there. Something analogous to https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0074-didcomm-best-practices/README.md.

2. `thread_id` needs to be consistent with the parent thread id field. So if we go with `thread_id`, then we are also signing up for `parent_thread_id` as well.

Like I said, I don't care what answer we pick, particularly -- as long as we address the two items above.

**Comment by kdenhartog** [comment]
> We should probably articulate best practices for header names
+1 definitely!

>thread_id needs to be consistent with the parent thread id field.
+1 as well

FWIW, I prefer `thid` for no other reason than the pedantic reason of not needing to see my linter complain everytime I use snake_case.

**Comment by brianorwhatever** [comment]
`parent_thread_id` seems pretty bad to me so I am going to create a PR changing everything to `thid` and `pthid` and add a bit of the aries language re: names however this spec does feel a bit wordy already so I'll try to keep it brief.
